packages/boot/test/bootstrapTests/test-walletSurvivesZoeRestart.ts

```diff
@@ -0,0 +1,339 @@
+      collateralAvailable: { value: scale6(setup.auction.start.collateral) },
+      startCollateral: { value: scale6(setup.auction.start.collateral) },
+      startProceedsGoal: { value: scale6(setup.auction.start.debt) },
+    });
+
+    console.log(collateralBrandKey, 'step 3 of 10');
+    await advanceTimeBy(3, 'minutes');
+
+    console.log(collateralBrandKey, 'step 4 of 10');
+    await advanceTimeBy(3, 'minutes');
+    // XXX updates for bid1 and bid2 are appended in the same turn so readLatest gives bid2
+    // NB: console output shows 8897786n payout which matches spec 8.897ATOM
+    // t.like(readLatest('published.wallet.agoric1buyer'), {
+    //   status: {
+    //     id: `${collateralBrandKey}-bid1`,
+    //     payouts: {
+    //       Bid: { value: 0n },
+    //       Collateral: { value: scale6(outcome.bids[0].payouts.Collateral) },
+    //     },
+    //   },
+    // });
+
+    t.like(readLatest('published.wallet.agoric1buyer'), {
+      status: {
+        id: `${collateralBrandKey}-bid2`,
+        payouts: likePayouts(outcome.bids[1].payouts),
+      },
+    });
+
+    console.log(collateralBrandKey, 'step 5 of 10');
+    await advanceTimeBy(3, 'minutes');
+
+    console.log(collateralBrandKey, 'step 6 of 10');
+    await advanceTimeBy(3, 'minutes');
+    t.like(readLatest(`published.auction.book${managerIndex}`), {
+      collateralAvailable: { value: 9659301n },
+    });
+
+    console.log(collateralBrandKey, 'step 7 of 10');
+    await advanceTimeBy(3, 'minutes');
+
+    console.log(collateralBrandKey, 'step 8 of 10');
+    await advanceTimeBy(3, 'minutes');
+
+    console.log(collateralBrandKey, 'step 9 of 10');
+    await advanceTimeBy(3, 'minutes');
+    // Not part of product spec
+    t.like(readLatest(metricsPath), {
+      numActiveVaults: 0,
+      numLiquidationsCompleted: setup.vaults.length,
+      numLiquidatingVaults: 0,
+      retainedCollateral: { value: 0n },
+      totalCollateral: { value: 0n },
+      totalCollateralSold: { value: 35340699n },
+      totalDebt: { value: 0n },
+      totalOverageReceived: { value: 0n },
+      totalProceedsReceived: { value: 309540000n },
+      totalShortfallReceived: { value: 0n },
+    });
+
+    console.log(collateralBrandKey, 'step 10 of 10');
+    // continuing after now would start a new auction
+    {
+      const { nextDescendingStepTime, nextStartTime } = readLatest(
+        'published.auction.schedule',
+      ) as Record<string, import('@agoric/time/src/types.js').TimestampRecord>;
+      t.is(nextDescendingStepTime.absValue, nextStartTime.absValue);
+    }
+
+    // bid3 still live because it's not fully satisfied
+    const { liveOffers } = readLatest('published.wallet.agoric1buyer.current');
+    t.is(liveOffers[0][1].id, `${collateralBrandKey}-bid3`);
+
+    // restart Zoe
+    // /////// Upgrading ////////////////////////////////
+    const zoeUpgradeSpec = {
+      package: 'builders',
+      packageScriptName: 'build:null-upgrade-zoe-proposal',
```

Comment:
I think @iomekam developed a way that we don't have to add a package script. Oh I see it's in review: https://github.com/Agoric/agoric-sdk/pull/8449/  

This is okay to land as is.